satpy/readers/yaml_reader.py

```diff
@@ -870,7 +870,114 @@ def _load_area_def(dsid, file_handlers):
     return final_area.squeeze()
 
 
-class GEOSegmentYAMLReader(FileYAMLReader):
+def _set_orientation(dataset, upper_right_corner='native', **kwargs):
+    """Set the orientation of datasets.
+
+    Allows to flip geostationary datasets without having to resample.
+
+    Args:
+        datasets (iterable): Limit the flipping to this list of names (str), wavelengths (float), or
+                            DatasetID objects.
+                            By default all currently loaded datasets are flipped.
+        upper_right_corner (str): Direction of the upper right corner of the image.
+                                Possible options are 'NW', 'NE', 'SW', 'SE', or 'native'.
+                                The common upright image orientation corresponds to 'NW'.
+                                Defaults to 'native' (no flipping is applied).
+
+    """
+    def _change_area_extent_of_areadef(areadef, new_extent):
+        """Change the area extent of an area definition and regenerate it."""
+        # does this need to be moved somewhere else?
+        return AreaDefinition(
+            areadef.area_id,
+            areadef.description,
+            areadef.proj_id,
+            areadef.proj_dict,
+            areadef.width,
+            areadef.height,
+            new_extent
+        )
+
+    if upper_right_corner == 'native':
+        logger.debug("Requested Dataset orientation is 'native'. No flipping is applied.")
+        return
+    if upper_right_corner not in ['NW', 'NE', 'SE', 'SW', 'native']:
+        raise ValueError("Origin corner not recognized. Should be 'NW', 'NE', 'SW', 'SE' or 'native.")
+
+    # get the target orientation
+    target_northup = False
+    target_eastright = False
+    if upper_right_corner in ['NW', 'NE']:
+        target_northup = True
+    if upper_right_corner in ['NW', 'SW']:
+        target_eastright = True
+
+    if 'area' not in dataset.attrs:
+        logger.info("Dataset is not a geographical dataset and cannot be flipped.")
+        return dataset
+
+    if dataset.attrs['area'].proj_dict['proj'] != 'geos':
```

Comment:
Is is possible to check the area's CRS here instead of the proj_dict?

Reply:
I guess so yes. What is the advantage of doing so?

Reply:
we're slowly migrating to using crs more instead of proj dicts

Reply:
also this here doesn't cover the projections defined by epsg code

Reply:
Understood. So I changed to:
```
if dataset.attrs['area'].crs.coordinate_operation.method_name not in ['Geostationary Satellite (Sweep Y)',
                                                                                                               'Geostationary Satellite (Sweep X)']:
```

Reply:
:+1: @mraspaud Can we be sure that the area def has a `crs` property or do we need to fallback to the proj_dict?

Reply:
Iirc the crs is a property of the areadef so always there

Reply:
`crs` should always exist *if* they are using a new-ish version of pyproj (2.0+).